p2p/discovery/message.go

```diff
@@ -18,39 +20,189 @@ const (
 	NeighborsMsg MsgType = 5
 )
 
-type Ping struct {
+const (
+	VERSION uint = 1
+)
+
+type ping struct {
 	Version uint // TODO add version check
-	ID      NodeID
+	SelfID  NodeID
+
+	to NodeID
 }
 
-type Pong struct {
-	ID NodeID
+type pong struct {
+	SelfID NodeID
 }
 
-type FindNode struct {
-	Target NodeID
+type findNode struct {
+	SelfID  NodeID
+	QueryID NodeID
+
+	to NodeID
 }
 
-type Neighbors struct {
-	Nodes []RPCNode
+type neighbors struct {
+	SelfID NodeID
+	Nodes  []*rpcNode
 }
 
-type RPCNode struct {
-	ID      NodeID
+type rpcNode struct {
+	SelfID  NodeID
 	IP      net.IP
 	UDPPort uint16
```

Comment:
so, UDPPort means that underlying protocol of RPC call is UDP, 
can we use TCP for RPC?

Reply:
for now we only use UDP